packages/ui/src/elements/WhereBuilder/index.tsx

```diff
@@ -144,9 +147,21 @@ export const WhereBuilder: React.FC<WhereBuilderProps> = (props) => {
   React.useEffect(() => {
     if (shouldUpdateQuery) {
       handleWhereChange({ or: conditions })
+
+      // Reset the page param when updating the where query
+      if (searchParams?.page && searchParams?.page !== '1') {
```

Comment:
This might fix the symptoms of this issue but I'm not sure this addresses the root cause of the problem here. Couple of things I'm noticing: there's already a `handleWhereChange` callback, feels like we can tie into that instead. This will ensure that you can override this behavior based on the environment that the `WhereBuilder` is rendered in, i.e. querying within a `ListDrawer` should _not_ effect top-level URL params. Take a look at the `ListQueryProvider`, feels like this is where the magic should happen. On the `ListView`, we're rendering it with the `modifySearchParams` prop. This is an existing function that interfaces with the URL. You should be able to adjust this as needed.

Reply:
On it!